events/core/types.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"time"
+
+	kube_api "k8s.io/kubernetes/pkg/api"
+)
+
+type EventBatch struct {
+	// When this batch was created.
+	Timestamp time.Time
+	// List of events included in the batch.
+	Events []*kube_api.Event
+}
+
+// A place from where the events should be scraped.
+type EventSource interface {
+	// This is a mutable method. Each call to this method clears the internal buffer so that
+	// each event can be obtained only once.
+	GetNewEvents() *EventBatch
+}
+
+type EventSink interface {
+	Name() string
+
+	// Exports data to the external storge. The funciton should be synchronous/blocking and finish only
+	// after the given EventBatch was written. This will allow sink manager to push data only to these
+	// sinks that finished writing the previous data.
+	ExportData(*EventBatch)
+	Stop()
```

Comment:
Please add comment on assumed semantics - I guess that it lets in-flight ExportData to finish before killing the Sink (in case someone would want to implement Stop() by os.Exit(0))